service/controller/v11/resource/deployment/master_deployment.go

```diff
@@ -89,11 +90,15 @@ func newMasterDeployments(customObject v1alpha1.KVMConfig) ([]*extensionsv1.Depl
 				Template: apiv1.PodTemplateSpec{
 					ObjectMeta: apismetav1.ObjectMeta{
 						GenerateName: key.MasterID,
+						Finalizers: []string{
+							key.DrainingNodesFinalizer,
+						},
```

Comment:
We manage a specific finalizer in the kvm-operator to have control over the guest cluster nodes deletion process.